### PR TITLE
Add ShoppingAssistant, and AlloyDB support to ProductCatalog

### PR DESCRIPTION
**Pull request by bourgeoisor**
This PR:
- Adds a new service (ShoppingAssistantService) which is disabled by default
  - Augmented the Frontend with a new chat bot assistant page where you can upload a photo of a room and ask for product recommendations
  - This calls upon the new service, which uses a combination of calls to Gemini and RAG (using AlloyDB vectors) to return a well-formatted prompt response
- Adds AlloyDB support for the ProductCatalogService (continuing to use local file by default)
- Fixes permission flags on files (adding / removing +x as needed)

### CODE REVIEW

**Review by NimJay — APPROVED** [review]
Approved. Thanks again, @bourgeoisor, for being very intentional about variable/file naming and modularizing (where necessary)! That made this code review really easy. :)

Please address any comments that start "**Issue:**" or "**Suggestion:**".

Overall, we just have to make sure we don't break the default deployment.
But I realize that the `main` branch is currently broken, we I've opened: https://github.com/GoogleCloudPlatform/microservices-demo/pull/2467

File: src/recommendationservice/recommendation_server.py
[the remark this answers is not in the document]
Comment (bourgeoisor) [reply src/recommendationservice/recommendation_server.py:1]: Yep a bit messy, but TL;DR:
- Removing the +x flag from files that don't need it (like images)
- Adding the +x flag to file that need it (like scripts)

File: src/productcatalogservice/catalog_loader.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+			return err
+		}
+		categories = strings.ToLower(categories)
+		product.Categories = strings.Split(categories, ",")
+
+		catalog.Products = append(catalog.Products, product)
+	}
+
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (bourgeoisor) [reply src/productcatalogservice/catalog_loader.go:151]: Yep, currently running live in a deployment.

File: src/shoppingassistantservice/.gitignore
[the remark this answers is not in the document]
Comment (bourgeoisor) [reply src/shoppingassistantservice/.gitignore:1]: Done on all counts!

File: kustomize/components/disable-shopping-assistant/kustomization.yaml
[the remark this answers is not in the document]
Comment (bourgeoisor) [reply kustomize/components/disable-shopping-assistant/kustomization.yaml:1]: Created: https://github.com/GoogleCloudPlatform/microservices-demo/issues/2468

**Pull request merged**